### PR TITLE
Gra 824 join refactor

### PR DESCRIPTION
**Pull request by mattkasun**
updates to new node/host/server struct....

what works:
join via username/network/server or token (single and dual stack networks)
list
install
version
connect/disconnect (with limitations)

what doesn't
leave
uninstall
pull 
 -- cannot authenticate with server due to way passwds are stored on server

daemon
  - daemon starts but wg-peers are not populated.

### CODE REVIEW

**Comment by mattkasun** [comment]
with refactor of peerupdates in netmaker; leave, uninstall, pull now work

File: functions/join.go [hunk 1]
```diff
@@ -65,31 +66,25 @@ func Join(flags *viper.Viper) error {
 		flags.Set("apiconn", accessToken.APIConnString)
 	}
 	logger.Log(1, "Joining network: ", flags.GetString("network"))
-	node, newServer, newHost, err := JoinNetwork(flags)
+	node, server, err := JoinNetwork(flags)
 	if err != nil {
 		return err
 	}
+	log.Println("server response to join")
```
Comment (0xdcarns) [inline functions/join.go:73]: debug logs?

File: wireguard/iface.go [hunk 2]
```diff
@@ -24,15 +24,14 @@ func IfaceExists(ifacename string) bool {
 // IfaceDelta - checks if the new node causes an interface change
 func IfaceDelta(currentNode *config.Node, newNode *config.Node) bool {
 	// single comparison statements
-	if newNode.EndpointIP.String() != currentNode.EndpointIP.String() ||
-		//newNode.PublicKey != currentNode.PublicKey ||
-		newNode.Address.IP.String() != currentNode.Address.IP.String() ||
+	if
+	//newNode.PublicKey != currentNode.PublicKey ||
+	newNode.Address.IP.String() != currentNode.Address.IP.String() ||
```
Comment (0xdcarns) [inline wireguard/iface.go:29]: move if to this line

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: functions/join.go (hunk 1 above)
Comment (mattkasun) [reply functions/join.go:73]: Yes, and for now they should stay.  Added to netclient cleanup

File: wireguard/wireguard_linux.go [hunk 3]
```diff
@@ -93,6 +94,7 @@ func (nc *NCIface) ApplyAddrs() error {
 		}
 	}
 	for _, node := range config.GetNodes() {
+		log.Println("adding address to wg interface", node.Address, node.Address6)
```
Comment (0xdcarns) [inline wireguard/wireguard_linux.go:97]: was this solved, or still required?

**Review by 0xdcarns — APPROVED** [review]

File: wireguard/wireguard_linux.go (hunk 3 above)
Comment (mattkasun) [reply wireguard/wireguard_linux.go:97]: not yet,  same ip is still assigned to all nodes

**Pull request merged**